features/toolbar/actions.js

```diff
@@ -46,15 +47,19 @@ export function hangup() {
                 .then(() => conference.leave());
         }
 
-        promise = promise
-            .then(() => dispatch(disposeLocalTracks()));
-
         if (connection) {
             promise = promise
                 .then(() => connection.disconnect());
         }
 
-        return promise;
+        // XXX Local tracks and local participant might exist without conference
+        // and connection initialized, so we need to explicitly clean them here.
+        // Furthermore, currently local tracks can be initialized before local
+        // participant is created, so we cannot hope that they will be destroyed
+        // when (and if) local participant leaves.
+        return promise
+            .then(() => dispatch(destroyLocalTracks()))
```

Comment:
OK, I suggest to do the following:

1. To dispose & remove local tracks we have ```destroyLocalTracks()``` action which is opposite to ```createLocalTrack()```. We should call it at the end of ```hangup()``` action after we left conference and disconnected from connection.

2. ```hangup()``` action seems to be an opposite to ```init()``` action of _"features/base/connection"_. I suggest to (possibly) rename to ```destroy()``` and move it from _"features/toolbar"_ to _"features/base/connection"_.

3. We can remove ```localParticipantLeft()``` action at all. It seems that local participant is created when we have conference and removed when conference is left, so we can remove local participant from Redux state on ```CONFERENCE_LEFT``` action inside participants' reducer. 

4. Yes, we can remove only remote tracks from Redux state inside tracks' reducer on ```PARTICIPANT_LEFT```. But if we will merge https://github.com/jitsi/lib-jitsi-meet/pull/174, then we can get rid of handling ```PARTICIPANT_LEFT``` inside tracks' reducer at all, because ```TRACK_REMOVED``` will be fired for tracks of leaving participant.